.github/workflows/community_ci.yml

```diff
@@ -62,68 +30,148 @@ jobs:
-        shell: bash
-        if: github.event_name == 'workflow_dispatch'
-        run: echo "branch=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
-        id: extract_branch
-      - name: Fetch last commit id from remote branch [PULL REQUESTS]
-        if: github.event_name == 'pull_request_target'
+
+      # On checkout the head commit is a detached head
+      # We have to use the git CLI to get the last commit id of the head branch
+      # We also perform validation and sanitization on the GITHUB_HEAD_REF to prevent command injection
+      - name: Fetch last commit id from remote branch
         id: fetch_last_commit_id_pr
-        run: echo "commit_id=$(git ls-remote --heads origin ${{ github.head_ref }} | cut -f 1)" >> $GITHUB_OUTPUT
-      - name: Fetch last commit id from remote branch [WORKFLOW DISPATCH]
-        if: github.event_name == 'workflow_dispatch'
-        id: fetch_last_commit_id_wd
-        run: echo "commit_id=$(git rev-parse origin/${{ steps.extract_branch.outputs.branch }})" >> $GITHUB_OUTPUT
-      - name: Test connectors [WORKFLOW DISPATCH]
-        if: github.event_name == 'workflow_dispatch'
+        run: |
+
+          # Validation: Check if GITHUB_HEAD_REF contains only alphanumeric characters, hyphens, underscores or slashes
+          if [[ ! "$GITHUB_HEAD_REF" =~ ^[a-zA-Z0-9_/-]+$ ]]; then
+            echo "Invalid input: '${GITHUB_HEAD_REF}'. Only alphanumeric characters, hyphens, underscores or slashes are allowed."
+            exit 1
+          fi
+
+          # Sanitization: Remove any characters that are not alphanumeric, hyphens, underscores, or slashes
+          sanitized_head_ref=$(echo "$GITHUB_HEAD_REF" | tr -cd '[:alnum:]-_/')
+          echo "commit_id=$(git ls-remote --heads origin '${sanitized_head_ref}' | cut -f 1)" >> $GITHUB_OUTPUT
+          echo $sanitized_head_ref
+
+      # This will sync the .github folder of the main repo with the fork
+      # This allows us to use up to date actions from the main repo
+      - name: Pull .github folder from main repository
+        id: pull_github_folder
+        run: |
+          git remote add main https://github.com/airbytehq/airbyte.git
+          git fetch main ${MAIN_BRANCH_NAME}
+          git checkout main/${MAIN_BRANCH_NAME} -- .github
+
+
+      - name: Run airbyte-ci format check all
+        # This path refers to the fork .github folder.
+        # We make sure its content is in sync with the main repo .github folder by pulling it in the previous step
         uses: ./.github/actions/run-airbyte-ci
         with:
-          context: "manual"
-          dagger_cloud_token: ${{ secrets.DAGGER_CLOUD_TOKEN }}
+          # TODO remove this binary URL before merging
+          airbyte_ci_binary_url: https://storage.googleapis.com/dev-airbyte-cloud-connector-metadata-service/airbyte-ci/releases/ubuntu/3570293/airbyte-ci
+          context: "pull_request"
+          dagger_cloud_token: ${{ secrets.DAGGER_CLOUD_TOKEN_2 }}
```

Comment:
Should we skip the token if it speeds things up? Are dagger caches actually getting read from if we run that CI on the _next_, not the same commit?

Reply:
> Are dagger caches actually getting read from if we run that CI on the next, not the same commit?

Yes they are. Dagger caches are docker layers and volume cache. For instance:
* If you did not change the dependencies of a connector they will get pulled from the poetry/pip cache volume
* If you modify two connectors on the branch but just made a change to one one your latest commit the cache will be re-used

Reply:
We could create a community specific dagger token